infra/recipes/terraform/azure/environments/sandbox/main.tf

```diff
@@ -1,35 +1,82 @@
-resource "azurerm_resource_group" "openhouse_sandbox" {
-    name = var.resource_group_name
-    location = var.resource_group_location
+data "azurerm_resource_group" "openhouse_sandbox" {
+  name = var.resource_group_name
+}
+
+data "azurerm_storage_account" "default" {
+  depends_on          = [module.storage]
+  resource_group_name = var.resource_group_name
+  name                = local.storage_account_name
+}
+
+resource "random_string" "storage_name" {
+  length  = 5
+  special = false
+  lower   = true
+  upper   = false
+}
+
+locals {
+  storage_account_name = "openhousestorage${random_string.storage_name.result}" // added random string of numbers to make it unique
+  container_name       = "blobcontainer"
+  db_username          = "azureadmin"
+  db_password          = "Pa33word"
+  db_name              = "openhouse-sandbox-db"
+  db_server_name       = "openhouse-sandbox-mysql-server"
 }
 
 module "vm" {
-    source = "../../modules/vm"
-    virtual_network_name = "openhouse-sandbox-network"
-    resource_group_name = var.resource_group_name
-    subnet_name = "openhouse-sandbox-subnet"
+  source               = "../../modules/vm"
+  depends_on           = [data.azurerm_resource_group.openhouse_sandbox]
+  virtual_network_name = "openhouse-sandbox-network"
+  resource_group_name  = var.resource_group_name
+  subnet_name          = "openhouse-sandbox-subnet"
 }
 
 module "mysql" {
-    source = "../../modules/mysql"
-    subnet_id = module.vm.subnet_id
-    resource_group_name = azurerm_resource_group.openhouse_sandbox.name
-    server_name = "openhouse-sandbox-mysql-server"
-    db_admin_login = "azureadmin"
-    db_admin_password = "Pa33word"
-    db_name = "openhouse-sandbox-db"
+  source              = "../../modules/mysql"
```

Comment:
We can use terraform `root.path` here I guess like ${abspath(path.root)}

Reply:
And we can get into the module directory from root path.

Reply:
yes sounds good

Reply:
Nevermind, variables are not allows in the module source. https://repost.aws/questions/QUKJXBzAHSR2qkNNLQAsXPDA/variable-in-terraform-module-source. I will add an absolute path still though